internal/npm_install/generate_build_file.js

```diff
@@ -317,15 +339,57 @@ function flattenDependencies(pkg, dep, pkgsMap) {
  */
 function printJson(pkg) {
   // Clone and modify _dependencies to avoid circular issues when JSONifying
+  // & delete _files array
   const cloned = {...pkg};
   cloned._dependencies = cloned._dependencies.map(dep => dep._dir);
+  delete cloned._files;
   return JSON.stringify(cloned, null, 2).split('\n').map(line => `# ${line}`).join('\n');
 }
 
+/**
+ * A filter function for a bazel filegroup.
+ * @param files array of files to filter
+ * @param exts list of white listed extensions; if empty, no filter is done on extensions;
+ *             '' empty string denotes to allow files with no extensions, other extensions
+ *             are listed with '.ext' notation such as '.d.ts'.
+ */
+function filterFilesForFilegroup(files, exts = []) {
+  files = files
+              // Bazel target names must be composed entirely of characters drawn
+              // from the set a–z, A–Z, 0–9, and the punctuation symbols _/.+-=,@~.
+              // Files with spaces are also not allowed in Bazel runfiles
+              // See https://github.com/bazelbuild/bazel/issues/4327
+              .filter(f => !f.match(/[^a-zA-Z0-9\_\/\.\+\-\=\,\@\~]/))
```

Comment:
@gregmagolan thanks for the patch!

could you please add `$` as valid symbol? not having it breaks [adv@5.5.2](https://github.com/epoberezkin/ajv/blob/v5.5.2/lib/%24data.js) because `$data` module is not symlinked into `bazel`'s runfiles tree

```suggestion
              .filter(f => !f.match(/[^a-zA-Z0-9\_\/\.\+\-\=\,\@\~\$]/))
```

Reply:
Thanks for pointing that out. I think the regex needs to be permissive. Will figure out exactly which characters are legal tomorrow.